extra/release_info_generator.py

```diff
@@ -0,0 +1,131 @@
+        else 365
+    )
+    expirationDate = tag.datetime + aYear
+    if currentMinor in releases:
+        eolDate = releases[currentMinor]["supported_until"].split("-")
+        expirationDate = datetime.fromisoformat(
+            f"{eolDate[0]}-{eolDate[1]}-{monthrange(int(eolDate[0]), int(eolDate[1]))[1]}T{str(time.max)}"
+        )
+    isExpired = expirationDate < endOfToday
+    return LtsEnd(isExpired, expirationDate)
+
+
+def collect_release_info(tag, minorRelease):
+    args = {
+        "list": "git",
+        "list_format": "json",
+        "in_integration_version": tag.version,
+    }
+    repoList = StringIO()
+    with redirect_stdout(repoList):
+        do_list_repos(
+            Namespace(**args),
+            False,
+            False,
+            False,
+        )
+
+    versionReposInfo = json.loads(repoList.getvalue())
+    existingInfo = {}
+    if tag.version in minorRelease:
+        existingInfo = minorRelease[tag.version]
+    result = {
+        "release_date": tag.datetime.strftime("%Y-%m-%d"),
+        **existingInfo,
+        **versionReposInfo,
+    }
+    result["repos"] = list(filter(lambda repo: (repo["version"]), result["repos"]))
+    return result
+
+
+def get_releases():
+    tags = []
+    saasTags = []
+    with open("tags", "r") as tagsFile:
+        for line in tagsFile:
+            tagMatch = tagMatcher.match(line.strip())
+            if not tagMatch:
+                continue
+            tag = Tag(tagMatch.group(1), datetime.fromtimestamp(int(tagMatch.group(2))))
+            if tag.version.startswith("saas"):
+                saasTags.append(
+                    {"tag": tag.version, "date": tag.datetime.strftime("%Y-%m-%d")}
+                )
+            match = releaseMatcher.fullmatch(tag.version)
+            if match and tag.version >= lastChecked:
+                tags.append(tag)
+    return {"saasTags": saasTags, "tags": tags}
+
+
+taggedReleases = get_releases()
+releases = taggedReleases["tags"]
+releaseInformation = {"lts": [], "releases": {}, "saas": taggedReleases["saasTags"]}
+
+existingReleaseInformation = {}
+try:
+    with open("versions.json", "r") as current:
+        existingReleaseInformation = json.load(current)
+except:
+    existingReleaseInformation = existingReleaseInformation
+if existingReleaseInformation:
+    releaseInformation = existingReleaseInformation
+
+ltsReleases = []
+for release in releases:
+    print(f"processing {release.version}")
+    match = releaseMatcher.match(release.version)
+    currentMinor = f"{match.group(1)}.{match.group(2)}"
+    initialMinorRelease = next(
+        (rel for rel in releases if rel.version == f"{currentMinor}.0"), release
+    )
+    ltsEnd = determine_lts_end(
+        initialMinorRelease, currentMinor, releaseInformation["releases"]
+    )
+    if ltsEnd and not ltsEnd.isExpired and currentMinor not in ltsReleases:
+        ltsReleases.append(currentMinor)
+    if currentMinor not in releaseInformation["releases"]:
```

Comment:
I take it this is the magic line which prevents old entries from being updated?

Reply:
yes, for all the information not sourced from the `release_tool` (there is also [this part](https://github.com/mendersoftware/integration/pull/1977/files#diff-e2e57cbd86e3596e05d64332a47f53983a12ea4fdd664cd2b9378af89e09f4c2R70-R74) that should ensure some extra information is also kept)